src/librbd/ExclusiveLock.cc

```diff
@@ -134,11 +134,13 @@ void ExclusiveLock<I>::shut_down(Context *on_shut_down) {
   {
```

Comment:
@dillaman, I modified it, and it works well, no issues were found yet, please have a look after the checks